### PR TITLE
Resource loading

### PR DESCRIPTION
**Pull request by StephaneDelcroix**
### Description of Change ###

More generic resourceLoading mechanism for the Previewer, so the same API can be reused to load different kind of resources.

### Bugs Fixed ###

/

### API Changes ###

/

### Behavioral Changes ###

/

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Build.Tasks/XamlCTask.cs [hunk 1]
```diff
@@ -195,6 +195,8 @@ public override bool Execute(out IList<Exception> thrownExceptions)
 							Logger.LogLine(2, "done");
 						}
 
+						Logger.LogLine(2, "");
```
Comment (alanmcgovern) [inline Xamarin.Forms.Build.Tasks/XamlCTask.cs]: Is this a debugging helper?

File: Xamarin.Forms.Xaml/XamlLoader.cs [hunk 2]
```diff
@@ -127,10 +131,14 @@ static void Visit (RootNode rootnode, HydratationContext visitorContext)
...
 			//the Previewer might want to provide it's own xaml for this... let them do that
-			if (Xamarin.Forms.Xaml.Internals.XamlLoader.XamlFileProvider != null && (xaml = Xamarin.Forms.Xaml.Internals.XamlLoader.XamlFileProvider(type)) != null)
+			//the check at the end is preferred (using ResourceLoader). keep this until all the previewers are updated
+
+#pragma warning disable 0618
+			var xaml = Internals.XamlLoader.XamlFileProvider?.Invoke(type);
+#pragma warning restore 0618
+
+			if (xaml != null && ResourceLoader.ResourceProvider != null)
 				return xaml;
```
Comment (alanmcgovern) [inline Xamarin.Forms.Xaml/XamlLoader.cs]: Why are you checking that `ResourceProvider != null` here? I presume the idea is that we will *either* be using `XamlFileProvider` *or* we'll be using `ResourceLoader.ResourceProvider` which means if xaml is non-null then ResourceProvider will always be null.

File: Xamarin.Forms.Build.Tasks/XamlCTask.cs (hunk 1 above)
Comment (StephaneDelcroix) [inline Xamarin.Forms.Build.Tasks/XamlCTask.cs]: no, it's not. adding that white line makes the build output clearer

File: Xamarin.Forms.Xaml/XamlLoader.cs (hunk 2 above)
Comment (StephaneDelcroix) [inline Xamarin.Forms.Xaml/XamlLoader.cs]: this `ResourceProvider != null` check is not really required, it's only make sure that if you were foolish enough to set both, the ResourceProvider will prevail.

Comment (alanmcgovern) [inline Xamarin.Forms.Xaml/XamlLoader.cs]: I think you've just hit the nail on the head - if you set both, and XamlFileProvider does give valid data, then ResourceProvider will *not* prevail. Was that expected?

Comment (StephaneDelcroix) [inline Xamarin.Forms.Xaml/XamlLoader.cs]: 🤦‍♂️

Comment (StephaneDelcroix) [inline Xamarin.Forms.Xaml/XamlLoader.cs]: this is a case when I should rebase to hide my embarrassment

**Pull request merged**